### PR TITLE
Delete coverage.sh

### PR DESCRIPTION
**Pull request by liamappelbe**
Removed the duplicate test step from the workflows, which speeds things up. ffigen's mac tests (with all the mac-only ObjC tests, and with coverage) now run faster than the windows tests.

While I was there I fixed some flakiness in automated_ref_count_test

Fixes https://github.com/dart-lang/native/issues/1120

### CODE REVIEW

**Comment by liamappelbe** [comment]
Confirmed we have nice logs for test failures.

File: pkgs/ffigen/test/native_objc_test/automated_ref_count_test.dart [hunk 1]
```diff
@@ -7,6 +7,9 @@
 // Objective C support is only available on mac.
 @TestOn('mac-os')
 
+// This test is slightly flaky.
```
Comment (dcharkes) [inline pkgs/ffigen/test/native_objc_test/automated_ref_count_test.dart:10]: Document why it is flaky.

**Review by dcharkes — APPROVED** [review]
Thanks @liamappelbe ! 🚀

File: pkgs/ffigen/test/native_objc_test/automated_ref_count_test.dart (hunk 1 above)
Comment (liamappelbe) [reply pkgs/ffigen/test/native_objc_test/automated_ref_count_test.dart]: Done

**Pull request merged**